src/main/java/com/nvidia/spark/rapids/jni/CastStrings.java

```diff
@@ -152,6 +154,137 @@ public static ColumnVector fromIntegersWithBase(ColumnView cv, int base) {
     return new ColumnVector(fromIntegersWithBase(cv.getNativeView(), base));
   }
 
+  /**
+   * Trims and parses a timestamp string column with time zone suffix to a
+   * timestamp column.
+   * Use the default time zone if string does not contain time zone.
+   *
+   * Supports the following formats:
+   * `[+-]yyyy*`
+   * `[+-]yyyy*-[m]m`
+   * `[+-]yyyy*-[m]m-[d]d`
+   * `[+-]yyyy*-[m]m-[d]d `
+   * `[+-]yyyy*-[m]m-[d]d [h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   * `[+-]yyyy*-[m]m-[d]dT[h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   * 
+   * Spark supports the following zone id forms:
+   *   - Z - Zulu time zone UTC+0
+   *   - +|-[h]h:[m]m
+   *   - A short id, see
+   * https://docs.oracle.com/javase/8/docs/api/java/time/ZoneId.html#SHORT_IDS
+   *   - An id with one of the prefixes UTC+, UTC-, GMT+, GMT-, UT+ or UT-,
+   *     and a suffix in the formats:
+   *     - +|-h[h]
+   *     - +|-hh[:]mm
+   *     - +|-hh:mm:ss
+   *     - +|-hhmmss
+   *  - Region-based zone IDs in the form `area/city`, such as `Europe/Paris`
+   *
+   * Unlike Spark, Spark-Rapids currently does not support DST time zones.
+   *
+   * Note:
+   * - Do not support cast special strings(epoch now today yesterday tomorrow) to timestamp.
+   * Spark31x supports cast special strings while Spark320+ do not supports
+   * - Do not support DST time zones, throw ai.rapids.cudf.CudfException
+   *   if contains DST time zones.
+   *
+   * Example:
+   * input = [" 2023", "2023-01-01T08:00:00Asia/Shanghai "]
+   * ts = toTimestamp(input, "UTC", allowSpecialExpressions = true, ansiEnabled =
+   * false)
+   * ts is: ['2023-01-01 00:00:00', '2023-01-01T00:00:00']
+   *
+   * Example:
+   * input = ["2023-01-01T08:00:00 non-exist-time-zone"]
+   * In ANSI mode: throws IllegalArgumentException
+   * In non-ANSI mode: return null value
+   * 
+   * @param cv                      The input string column to be converted.
+   * @param defaultTimeZone         Use the default time zone if string does not
+   *                                contain time zone.
+   * @param ansiEnabled             is Ansi mode
+   * @return a timestamp column
+   * @throws IllegalArgumentException if any string in cv has invalid format or the time zone is
+   *                                  non-existed/wrong when ansiEnabled is true
+   * @throws CudfException            if time zone is a DST time zone
+   */
+  public static ColumnVector toTimestamp(ColumnView cv, ZoneId defaultTimeZone, boolean ansiEnabled) {
+    if (!GpuTimeZoneDB.isSupportedTimeZone(defaultTimeZone)) {
+      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
+              defaultTimeZone.toString()));
+    }
+
+    GpuTimeZoneDB singleton = GpuTimeZoneDB.getInstance();
```

Comment:
This feels like GpuTimeZoneDB is not very user friendly. Why is `getZoneIDMap` only part of the singleton? Why is it that we have to check if it is laded before we get the ZoneIdMap or the other pieces of information.

Reply:
Done, refactored `GpuTimeZoneDB `.
Provided more user friendly APIs.